src/api/app/models/project/key_info.rb

```diff
@@ -34,6 +36,7 @@ def self.find_by_project(project)
         key_info_params[:ssl_certificate] = parsed_response['sslcert']
       end
 
+      key_info_params.delete_if {|_key, value| value == project.name }
```

Comment:
Can we do something like this then? So that the condition only applies to the :origin
```ruby
hash.delete(:origin) if origin == project.name
```